docs/docs/shards/index.md

```diff
@@ -164,4 +164,596 @@ Showing how we mix the Clojure-like language as a template language
+    ```
+
+### None
+
+Type `None` indicates that no data type is expected. This implies that no value is expected.
+
+For example, `None` as one of the valid data types for `:Max` parameter in shard [`(RandomInt)`](https://docs.fragcolor.xyz/shards/General/RandomInt/) means that setting a value for this parameter is not mandatory.
+
+```clojure linenums="1"
+(RandomInt
+:Max [(None) (Int) (ContextVar [(Int)])]
+)
+```
+
+`(RandomInt)` generates a random integer and the `:Max` parameter is the upper limit (not inclusive) of the value that can be generated. So it makes sense to have `None` as one of the valid types for this `:Max` parameter for cases when you do not want an upper limit on the random integer.
+
+=== "Code"
+
+    ```clojure linenums="1"
+    (RandomInt 8)
+    (Log)   ;; max int that can be generated is 7
+    
+    (RandomInt)
+    (Log)   ;; no upper limit on the generated int
+    ```
+
+=== "Output"
+
+    ```
+    [info] [2022-07-22 13:45:03.282] [T-19992] [logging.cpp::55] [mywire] 4
+    [info] [2022-07-22 13:45:03.293] [T-19992] [logging.cpp::55] [mywire] 311828859
+    ```
+
+### Enum
+
+`Enum` stands for enumerated data type. 
+
+What this means is that the value you pass to an enumerated variable can only take certain 'states' or named constant values.
+
+For example, in [`(Math.Mean)`](https://docs.fragcolor.xyz/shards/Math/Mean/) the value for `:Kind` parameter needs to be of type `Enum`.
+
+```clojure linenums="1"
+(Math.Mean
+  :Kind [(Enum)]
+)
+```
+
+`(Math.Mean)` computes the mean of a sequence of floating-point numbers. But there are three kinds of means - Arithmetic mean, Geometric mean, and Harmonic mean.
+
+So the parameter `:Kind` is defined as an enum variable with these three fixed states :
+` :Kind = {Arithmtic mean, Geometric mean, Harmonic mean}`
+
+And hence `:Kind` expects a value that matches one of its possible states. In other words the value you pass in for `:Kind` needs to be an enumerated data type.
+
+In simple terms it just means that you pass in one of the allowed named constant values. Anything else will fail validation.
+
+=== "Code"
+
+    ```clojure linenums="1"
+    [2.0 10.0]
+    (Math.Mean :Kind Mean.Arithmetic)
+    (Log)   ;; AM => 6
+
+    [2.0 10.0]
+    (Math.Mean :Kind Mean.Geometric)
+    (Log)   ;; GM => 4.47214
+
+    [2.0 10.0]
+    (Math.Mean :Kind Mean.Harmonic)
+    (Log)   ;; HM => 3.33333
+    
+    [2.0 10.0]
+    (Math.Mean 
+    ;; AM is default, anything else will throw error
+        ;; :Kind 123
+        )   
+    (Log)   ;; AM => 6
+    ```
+
+=== "Output"
+
+    ```
+    [info] [2022-07-22 15:35:00.868] [T-15316] [logging.cpp::55] [mywire] 6
+    [info] [2022-07-22 15:35:00.881] [T-15316] [logging.cpp::55] [mywire] 4.47214
+    [info] [2022-07-22 15:35:00.882] [T-15316] [logging.cpp::55] [mywire] 3.33333
+    [info] [2022-07-22 15:35:00.883] [T-15316] [logging.cpp::55] [mywire] 6
+    ```
+
+### Bool
+
+Type `Bool` allows only two values - `true` or `false`.
```

Comment:
Should be lowercase.

```suggestion
Type `bool` allows only two values - `true` or `false`.
```

Reply:
Fixed in commit #[190ce35f2cccc0d0714284d4e929d35f5ee6c3f1](https://github.com/fragcolor-xyz/shards/pull/325/commits/190ce35f2cccc0d0714284d4e929d35f5ee6c3f1).